andela/ah-django-unchained-frontend
# #162948807 Implement rating of articles

**What does this PR do?**
Implements rating of articles by users.
- A user can rate other authors articles
- A user can view average ratings of a specific article

**Description of tasks to be completed**
- create a component to rate articles
- add actions and types for article rating
- add reducer for article rating
- add tests

**How should this be manually tested?**
Clone this repository on your local machine using
  `git clone https://github.com/andela/ah-django-unchained-frontend.git`

Cd into the project directory
  `cd ah-django-unchained-frontend`

Switch to the specific branch
  `git checkout ft-article-ratings-162948807`

Create a `.env` file and add `REACT_APP_BASE_URL=app-base-url` to the file.

Run `npm install` to install all the modules listed in the `package.json` file.

Run `npm start` command to start the application. The application should start successfully.

On the browser, navigate to the URL `http://localhost:3000/users` to signup a user.

Use URL `http://localhost:3000/article/slug` to fetch an article.

Click on any of the stars below the article to rate that article. Options are 1 to 5.

**Background Context**

- Rating option is hidden from an author if they are viewing their own article.
- Rating option is hidden from users that are not signed in.

**Screenshot**

View single article

![image](https://user-images.githubusercontent.com/32644295/53593630-35868c00-3baa-11e9-9b31-11350f1e97cd.png)


View all articles
![image](https://user-images.githubusercontent.com/32644295/53593402-ad07eb80-3ba9-11e9-879f-85f353f6f918.png)



**Relevant Pivotal Tracker stories**
[#162948807](https://www.pivotaltracker.com/story/show/162948807)
